sgv2-service-common/pom.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.stargate</groupId>
+        <artifactId>stargate</artifactId>
+        <version>1.0.44-SNAPSHOT</version>
+    </parent>
+    <artifactId>sgv2-service-common</artifactId>
+    <dependencies>
+        <dependency>
+            <groupId>com.github.misberner.duzzt</groupId>
+            <artifactId>duzzt-processor</artifactId>
+            <version>0.0.2</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.immutables</groupId>
+            <artifactId>value</artifactId>
+            <version>2.8.8</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
```

Comment:
It brings the `@Nullable` annotation that's used to mark field optional in `@Immutable` classes.

Reply:
Hm, ok, maybe then this could also be on the `provided` scope?

Furthermore, I bumped to this: https://github.com/immutables/immutables/issues/647#issuecomment-313467651, there are multiple options for avoiding non-null annotation (like optional, or this value check style).. Not sure if anything of that is better than this, wdyt?

Reply:
The value of non-null/nullable usually seems to be for document as not many frameworks (and few if any runtimes) make any use of it. On plus side I think it is fine to then leave dependency as `provided` since during runtime annotation dependencies are not strictly needed (unless some library/framework itself needs it & if so they should bring in the dependency).